io/net/virt-net/Network_virtualization.py

```diff
@@ -0,0 +1,311 @@
+        '''
+        p = pxssh.pxssh()
+        # Work-around for old pxssh not having options= parameter
+        p.SSH_OPTS = p.SSH_OPTS + " -o 'StrictHostKeyChecking=no'"
+        p.SSH_OPTS = p.SSH_OPTS + " -o 'UserKnownHostsFile /dev/null' "
+        p.force_password = True
+
+        p.login(ip, username, password)
+        p.sendline()
+        p.prompt(timeout=60)
+        # Ubuntu likes to be "helpful" and alias grep to
+        # include color, which isn't helpful at all. So let's
+        # go back to absolutely no messing around with the shell
+        p.set_unique_prompt()
+        p.prompt(timeout=60)
+        self.pxssh = p
+
+    def run_command(self, command, timeout=300):
+        '''
+        SSH Run command method for running commands on remote server
+        '''
+        self.log.info("Running the command on hmc %s", command)
+        c = self.pxssh
+        c.sendline(command)
+        c.expect("\n")  # from us
+        c.expect(c.PROMPT, timeout=timeout)
+        output = c.before.splitlines()
+        c.sendline("echo $?")
+        c.prompt(timeout)
+        return output
+
+    def rsct_service_start(self):
+        '''
+        Running rsct services which is necessary for Network
+        virtualization tests
+        '''
+        try:
+            for svc in ["rsct", "rsct_rm"]:
+                process.run('startsrc -g %s' % svc, shell=True, sudo=True)
+        except CmdError as details:
+            self.log.debug(str(details))
+            self.fail("Starting service %s failed", svc)
+
+        output = process.system_output("lssrc -a", ignore_status=True,
+                                       shell=True, sudo=True)
+        if "inoperative" in output:
+            self.cancel("Failed to start the rsct and rsct_rm services")
+
+    def test_network_virtualization_add(self):
+        '''
+        Network virtualized device add operation
+        '''
+        self.network_virtualization_add_remove_device(self.server, 'a',
+                                                      self.lpar_id,
+                                                      self.slot_num,
+                                                      self.backing_devices,
+                                                      'add')
+        output = self.network_virtualization_list_device(self.server,
+                                                         self.lpar,
+                                                         self.slot_num)
+        if 'slot_num=%s' % self.slot_num not in str(output):
+            self.log.debug(output)
+            self.fail("lshwres fails to list Network virtualized device \
+                       after add operation")
+
+    def test_network_virtualization_backing_device_add(self):
+        '''
+        Adding Backing device for Network virtualized device
+        '''
+        pre_add = self.backing_dev_count(self.slot_num)
+        self.backing_dev_add_remove(self.server, 'a', self.lpar_id,
```

Comment:
This should ideally be:
```
self.backing_dev_add_remove('a')
```
All other params are readable by the function also, so no need to pass them.